src/SonarScanner.MSBuild.Shim/ScannerEngineInput.cs

```diff
@@ -51,29 +45,27 @@ public ScannerEngineInput CloneWithoutSensitiveData()
         result.moduleKeys.UnionWith(moduleKeys);
         foreach (var property in scannerProperties)
         {
-            var key = property["key"].Value<string>();
-            var value = property["value"].Value<string>();
-            if (key == "sonar.modules")
-            {
-                result.modules.Value = value;
-            }
-            else
-            {
-                result.Add(key, ProcessRunnerArguments.ContainsSensitiveData(key) || ProcessRunnerArguments.ContainsSensitiveData(value) ? "***" : value);
-            }
+            var key = property.Key;
+            var value = property.Value;
+            result.Add(key, ProcessRunnerArguments.ContainsSensitiveData(key) || ProcessRunnerArguments.ContainsSensitiveData(value) ? "***" : value);
         }
         return result;
     }
 
     public override string ToString() =>
-        JsonConvert.SerializeObject(root, Formatting.Indented);
+        JsonConvert.SerializeObject(
+            new JObject
+            {
+                new JProperty("scannerProperties", new JArray(scannerProperties.Select(x => new JObject(new JProperty("key", x.Key), new JProperty("value", x.Value)))))
```

Comment:
Does this work? I think it reads a bit better. It should work as there is an implicit conversion from string to JToken:
```suggestion
                new JProperty("scannerProperties", new JArray(scannerProperties.Select(x => new JObject()
                { { "key", x.Key } },
                { { "value", x.Value } } )
```